hobbit-gui/gui-serverbackend/src/main/java/de/usu/research/hobbit/gui/rest/ExperimentsResources.java

```diff
@@ -65,28 +67,81 @@
         if (Application.isUsingDevDb()) {
             return getDevDb().queryExperiments(ids, challengeTaskId);
         } else {
+            // get user
+            UserInfoBean userInfo = InternalResources.getUserInfoBean(sc);
+            List<SystemBean> userSystems = PlatformControllerClientSingleton.getInstance().requestSystemsOfUser(userInfo.getPreferredUsername());
+            // create set of user owned system ids
+            String[] sysIds = userSystems.stream().map(s -> s.getId()).toArray(String[]::new);
+            Set<String> userOwnedSystemIds = new HashSet<>(Arrays.asList(sysIds));
```

Comment:
You could even put the requesting of user systems and the generating of the `Set` into an own method.